easycv/core/evaluation/ocr_eval.py

```diff
@@ -0,0 +1,282 @@
+                      recall) == 0 else 2.0 * precision * recall / (
+                          precision + recall)
+
+        matchedSum += detMatched
+        numGlobalCareGt += numGtCare
+        numGlobalCareDet += numDetCare
+
+        perSampleMetrics = {
+            'gtCare': numGtCare,
+            'detCare': numDetCare,
+            'detMatched': detMatched,
+        }
+        return perSampleMetrics
+
+    def combine_results(self, results):
+        numGlobalCareGt = 0
+        numGlobalCareDet = 0
+        matchedSum = 0
+        for result in results:
+            numGlobalCareGt += result['gtCare']
+            numGlobalCareDet += result['detCare']
+            matchedSum += result['detMatched']
+
+        methodRecall = 0 if numGlobalCareGt == 0 else float(
+            matchedSum) / numGlobalCareGt
+        methodPrecision = 0 if numGlobalCareDet == 0 else float(
+            matchedSum) / numGlobalCareDet
+        methodHmean = 0 if methodRecall + methodPrecision == 0 else 2 * methodRecall * methodPrecision / (
+            methodRecall + methodPrecision)
+        # print(methodRecall, methodPrecision, methodHmean)
+        # sys.exit(-1)
+        methodMetrics = {
+            'precision': methodPrecision,
+            'recall': methodRecall,
+            'hmean': methodHmean
+        }
+
+        return methodMetrics
+
+    def evaluate(self, preds, gt_polyons_batch, ignore_tags_batch, **kwargs):
+        results = []
+        for pred, gt_polyons, ignore_tags in zip(preds, gt_polyons_batch,
+                                                 ignore_tags_batch):
+            # prepare gt
+            gt_info_list = [{
+                'points': gt_polyon,
+                'text': '',
+                'ignore': ignore_tag
+            } for gt_polyon, ignore_tag in zip(gt_polyons, ignore_tags)]
+            # prepare det
+            det_info_list = [{
+                'points': det_polyon,
+                'text': ''
+            } for det_polyon in pred]
+            result = self.evaluate_image(gt_info_list, det_info_list)
+            results.append(result)
+        results = self.combine_results(results)
+        print(results)
+        return results
+
+
+@EVALUATORS.register_module()
+class OCRRecEvaluator(Evaluator):
+
+    def __init__(self,
+                 is_filter=False,
+                 ignore_space=True,
+                 dataset_name=None,
+                 metric_names=['acc']):
+        super().__init__(dataset_name, metric_names)
+        self.is_filter = is_filter
+        self.ignore_space = ignore_space
+        self.eps = 1e-5
+
+    def _evaluate_impl(self, predictions, labels, **kwargs):
+        pass
+
+    def _normalize_text(self, text):
+        text = ''.join(
+            filter(lambda x: x in (string.digits + string.ascii_letters),
+                   text))
+        return text.lower()
+
+    def evaluate(self, preds, labels, **kwargs):
+        correct_num = 0
```

Comment:
Why not put the implementation in evaluate into _evaluate_impl？

Reply:
done